src/Middleware/ResponseCompression/src/ResponseCompressionBody.cs

```diff
@@ -201,7 +201,7 @@ public override async ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, Cancella
             }
         }
 
-        private void InitializeCompressionHeaders()
+        private ICompressionProvider? InitializeCompressionHeaders()
```

Comment:
nit: maybe rename this GetCompressionProvider since now it returns the provider, maybe add a short comment saying why it would return null (compression not enabled, etc)

Reply:
Renaming it would be misleading, it's main job is to set the headers. I will add a comment about the return value.